src/api/codereader.rs

```diff
@@ -0,0 +1,259 @@
+use byteorder::{BigEndian, ReadBytesExt};
+use std::fs::File;
+use std::io::{Cursor, Read};
+use std::io::{Error, ErrorKind};
+use std::result::Result;
+use std::slice;
+
+use crate::objects::codeholder::CodeHolder;
+use crate::objects::instruction::Instruction;
+use crate::objects::register::{Register, RegisterLocation, RegisterReference};
+
+/// Creates a register instance from 5 bytes
+fn make_register(cur: &mut Cursor<&Vec<u8>>) -> Result<Register, Error> {
+    let reg = cur.read_u32::<BigEndian>()?;
+    let mut locval: u8 = 0;
+    cur.read(slice::from_mut(&mut locval))?;
+
+    let regloc = match locval {
+        0x01 => RegisterLocation::ConstantPool,
```

Comment:
For this and all other "magic numbers", it would be best to take advantage of the `const` keyword in Rust: https://doc.rust-lang.org/rust-by-example/custom_types/constants.html